src/bin/vip-import-validate-files.js

```diff
@@ -0,0 +1,153 @@
+import url from 'url';
+import path from 'path';
+import chalk from 'chalk';
+import fs from 'fs';
+import { promisify } from 'util';
+
+/**
+ * Internal dependencies
+ */
+import command from 'lib/cli/command';
+import {
+	acceptedExtensions,
+	findNestedDirectories,
+	folderStructureValidation,
+	isFileSanitized,
+	doesImageHaveExistingSource,
+	logErrorsForIntermediateImages,
+	logErrorsForInvalidFileTypes,
+	logErrorsForInvalidFilenames,
+} from '../lib/vip-import-validate-files';
+
+// Promisify to use async/await
+const stat = promisify( fs.stat );
+const readDir = promisify( fs.readdir );
+
+command( { requiredArgs: 1, format: true } )
+	.example( 'vip import validate files <file>', 'Validate your media files' )
+	.argv( process.argv, async ( arg, options ) => {
+		/**
+		 * File manipulation
+		 *
+		 * Manipulating the file path/name to extract the folder name
+		 */
+		const folder = arg.join(); // File comes in as an array as part of the args- turn it into a string
+		arg = url.parse( folder ); // Then parse the file to its URL parts
+		const filePath = arg.path; // Extract the path of the file
+
+		/**
+		 * Folder structure validation
+		 *
+		 * Find nested directories to see if media files follow the WordPress recommended folder structure
+		 *
+		 * Recommended structure: `uploads/year/month` (Single sites)
+		 */
+		const nestedDirectories = await findNestedDirectories( filePath );
+
+		if ( nestedDirectories ) {
+			folderStructureValidation( nestedDirectories );
+		}
+
+		/**
+		 * Individual file validations
+		 *
+		 * - Media file extension/type validation
+		 * - Filename validation
+		 * - Intermediate image validation
+		 */
+		let files;
+
+		try {
+			files = await readDir( nestedDirectories );
+
+			if ( ! files || ! files.length || files.length <= 0 ) {
+				console.error( chalk.red( '✕ Error:' ), 'Media files directory cannot be empty' );
+			}
+		} catch ( error ) {
+			console.error( chalk.red( '✕ Error:' ), `Unable to read directory ${ filePath }: ${ error.message }` );
+		}
+
+		/**
+		 * Media file extension validation
+		 *
+		 * Ensure that prohibited media file types are not used
+		 */
+
+		// Collect invalid files for error logging
+		const errorFileTypes = [];
+		const errorFileNames = [];
+		const intImagesObject = {};
+
+		// Iterate through each file to isolate the extension name
+		for ( const file of files ) {
+			// Check if file is a directory
+			const stats = await stat( nestedDirectories + '/' + file );
+			const isFolder = stats.isDirectory();
+
+			const extension = path.extname( file ); // Extract the extension of the file
+			const ext = extension.substr( 1 ); // We only want the ext name minus the period (e.g- .jpg -> jpg)
+			const extLowerCase = ext.toLowerCase(); // Change any uppercase extensions to lowercase
+
+			// Check for any invalid file extensions
+			// Returns true if ext is invalid; false if valid
+			const invalidExtensions = acceptedExtensions.indexOf( extLowerCase ) < 0;
```

Comment:
Consider using an `.includes` statement since you don't need the actual location.

Reply:
Got it, changed 👍